### PR TITLE
Add AGPL license

### PR DESCRIPTION
**Pull request by Xeeynamo**
From the [public announcement](https://discord.com/channels/1079389589950705684/1079395108501331990/1276310695964835910) on Discord:

> We've been progressing very far with the project and I started wondering what would be the best course of action to protect the work of existing and future contributors. I personally consider contributions to a decompilation project as creative work. Therefore it is my desire to protect this creative work from being re-claimed by other people. This is especially true in the light of the ability of modding the existing decompiled code and create derivative work.
> 
> I talked privately with the current most active contributors. And me, @sozud , @bismurphy and @joshlory so far agreed on pushing through the AGPLv3 license for all the decompiled code in `sotn-decomp/src`, excluding any third party code and decompiled code from the PSX SDK. But I've been wanting to ask all of you if you are fine your code to be licensed and protected under the AGPLv3 terms. 
>
> https://raw.githubusercontent.com/Xeeynamo/sotn-decomp/89686514916cabd0dd88ae0387f749a889c19e05/src/LICENSE
> 
> I put and all the contributors, which can be accessed to the page https://github.com/Xeeynamo/sotn-decomp/graphs/contributors . Again, this will work for existing and future contributors. In the license I put my full name as I will be responsible of any legality. And I understand not every contributor wants to disclose their identity. Please give an approval to my latest PR if you do not have any objection. If not, please write here publicly.
>
> PS. I understand past contributors will either not see the message above or decide to ignore it. If I do not hear from anyone of the contributors for more than one week between those who did not interact with me or the server, I will still go ahead and apply the license. It is not in my intention to ignore past contributors. But I do not want contributors who have been inactive for months or years to be a blocker. Please do not mistake this as an act of malevolence.

### CODE REVIEW

**Review by bismurphy — APPROVED** [review]

File: src/main/psxsdk/libapi/a07.c [hunk 1]
```diff
@@ -1,3 +1,4 @@
+// SPDX-License-Identifier: AGPL-3.0-only
```
Comment (sozud) [inline src/main/psxsdk/libapi/a07.c:1]: It looks like you aren't excluding psxsdk

File: src/pc/3rd/cJSON/cJSON.c [hunk 2]
```diff
@@ -1,3 +1,4 @@
+// SPDX-License-Identifier: AGPL-3.0-only
```
Comment (sozud) [inline src/pc/3rd/cJSON/cJSON.c:1]: We can't relicense this

Comment (Xeeynamo) [reply src/pc/3rd/cJSON/cJSON.c:1]: oopsie, I will exclude this folder too

File: src/saturn/lib/gfs.c [hunk 3]
```diff
@@ -1,3 +1,4 @@
+// SPDX-License-Identifier: AGPL-3.0-only
```
Comment (sozud) [inline src/saturn/lib/gfs.c:1]: src/saturn/lib should be excluded

File: src/LICENSE [hunk 4]
```diff
@@ -0,0 +1,15 @@
+// sotn-decomp, re-interpretation of the game Castlevania:Symphony of the Night
+// Copyright (C) 2024  Luciano Ciccariello and all the contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
```
Comment (sozud) [inline src/LICENSE:7]: The license doesn't match the tag you picked `AGPL-3.0-only`

Comment (Xeeynamo) [reply src/LICENSE]: adjusted and copied from https://spdx.org/licenses/AGPL-3.0-only.html

**Review by sozud — APPROVED** [review]

**Review by joshlory — APPROVED** [review]

**Review by SestrenExsis — APPROVED** [review]

**Review by hohle — APPROVED** [review]

**Review by NotExactlySiev — APPROVED** [review]

**Review by mkst — APPROVED** [review]

**Review by cram0 — APPROVED** [review]

**Comment by KiritoDv** [comment]
I suggest you should do something like this https://github.com/n64decomp/mk64/issues/474 since you need to every single one of the 30 contributors to accept the license change explicitly, and something personal, since this is a decomp i dont think that is the correct license, it should be cc0

**Comment by sozud** [comment]
> it should be cc0

CC0 doesn't seem to be OSI approved so I wouldn't recommend it https://opensource.org/licenses

**Review by synamaxmusic — APPROVED** [review]

**Comment by sozud** [comment]
@sonicdcer What licenses are you willing to accept?

**Review by DerpPrincess — APPROVED** [review]

**Comment by sozud** [comment]
I figure I should explain more why I think AGPLv3 makes sense vs. CC0 since there's a number of decomp projects that have chosen it or are working on re-licensing to it.

https://github.com/sonicdcer/sf64?tab=CC0-1.0-1-ov-file#readme
https://github.com/n64decomp/mk64/issues/474
https://github.com/n64decomp/sm64?tab=CC0-1.0-1-ov-file

AGPLv3 is a strong copyleft license which means that people must release their modifications to the source. This keeps the project maximally open to the community so everyone can benefit from those modifications.

AGPLv3 has a disclaimer of warranty, and limits liability, which protects the contributors.

AGPLv3 has a patent license, protects users of the software from being sued for violating patents.

**Comment by sozud** [comment]
@ser-pounce @rossimo @frog404 Any issues with this change?

**Comment by ser-pounce** [comment]
@sozud all good, thanks for asking!

**Comment by rossimo** [comment]
I'm all good

**Comment by sozud** [comment]
Current status for `./src` (the part we are trying to re-license) is the following. This list isn't perfect since it includes psxsdk, which we aren't trying to re-license at this time. Generated with `git shortlog -n -s -- src`

```
*  821  Luciano Ciccariello
*  382  bismurphy
*  290  sozud
   159  Alejandro Asenjo Nitti
*   49  Jonathan Hohle
*   38  Josh Lory
*   32  ser-pounce
*   21  SestrenExsis
*   16  SynaMax
*   14  Julien Augugliaro
*   12  Allison Mackenzie Johnson
*    8  Mark Street
*    8  Ross Squires
     5  frog404
     4  Tuomas Tynkkynen
*    3  NotExactlySiev
     2  piratesephiroth
     1  Javier Córdoba Romero
     1  MottZilla
     1  Rainchus
     1  smb123w64gb
```


@dezgeg @piratesephiroth @kpanicc @MottZilla @Rainchus @smb123w64gb Any issues with this change?

**Comment by kpanicc** [comment]
All good, thank you for asking!
@sozud

**Comment by frog404** [comment]
All good for me too 👌
I feel it was an eternity ago that I did my modest PRs... You people are doing an amazing job !
Thanks for that and have a great day

**Comment by dezgeg** [comment]
Fine by me

**Comment by smb123w64gb** [comment]
No issue here.

**Comment by Xeeynamo** [comment]
@sonicdcer I kindly ask you to provide your feedback. For legal purposes we need a public record here.

**Comment by sozud** [comment]
@sonicdcer is the last major contributor that hasn't approved this PR. I talked to him directly to determine what he would need to approve this license change since I think reverting and redoing his commits is a lot of work that we should avoid if possible:

1. He requests that src/LICENSE either be removed since the root directory LICENSE (full AGPLv3) is sufficient in his opinion. His concern is that explicitly naming individuals takes away from the group nature of the project.

2. He wants to be unbanned from the sotn-decomp GitHub and Discord, and his Discord Contributor role restored.

**Comment by sozud** [comment]
sonicdcer asked me to delete a previous version of this comment since it included some info that wasn't pertinent to the license discussion, so I am re-posting:

He requests that src/LICENSE be removed since the root directory LICENSE (full AGPLv3) is sufficient in his opinion. His concern is that explicitly naming individuals takes away from the group nature of the project.

**Comment by Max-Clark** [comment]
AGPL sounds great to me 👍

**Comment by sonicdcer** [comment]
I agree to license all my contributions to AGPLv3.

**Review by sonicdcer — APPROVED** [review]

**Comment by Rainchus** [comment]
I agree to license my contributions with AGPLv3

**Comment by MottZilla** [comment]
> @dezgeg @piratesephiroth @kpanicc @MottZilla @Rainchus @smb123w64gb Any issues with this change?

I have no issues with the change. License is fine.

**Pull request merged**